linux_os/guide/services/ssh/ssh_server/sshd_set_maxstartups/oval/shared.xml

```diff
@@ -0,0 +1,71 @@
+    <ind:filepath operation="equals">/etc/ssh/sshd_config</ind:filepath>
+    <ind:pattern operation="pattern match" datatype="string">(?i)^\s*MaxStartups\s+(\d+):\d+:\d+\s*$</ind:pattern>
+    <ind:instance operation="greater than or equal" datatype="int">1</ind:instance>
+  </ind:textfilecontent54_object>
+  <ind:textfilecontent54_object id="obj_sshd_config_maxstartups_second_parameter" version="1">
+    <ind:filepath operation="equals">/etc/ssh/sshd_config</ind:filepath>
+    <ind:pattern operation="pattern match" datatype="string">(?i)^\s*MaxStartups\s+\d+:(\d+):\d+\s*$</ind:pattern>
+    <ind:instance operation="greater than or equal" datatype="int">1</ind:instance>
+  </ind:textfilecontent54_object>
+  <ind:textfilecontent54_object id="obj_sshd_config_maxstartups_third_parameter" version="1">
+    <ind:filepath operation="equals">/etc/ssh/sshd_config</ind:filepath>
+    <ind:pattern operation="pattern match" datatype="string">(?i)^\s*MaxStartups\s+\d+:\d+:(\d+)\s*$</ind:pattern>
+    <ind:instance operation="greater than or equal" datatype="int">1</ind:instance>
+  </ind:textfilecontent54_object>
+
+  <ind:textfilecontent54_state id="ste_sshd_config_start_parameter_valid" version="1">
+    <ind:subexpression datatype="int" operation="less than or equal">10</ind:subexpression>
+  </ind:textfilecontent54_state>
+  <ind:textfilecontent54_state id="ste_sshd_config_rate_parameter_valid" version="1">
+    <ind:subexpression datatype="int" operation="greater than or equal">30</ind:subexpression>
+  </ind:textfilecontent54_state>
+  <ind:textfilecontent54_state id="ste_sshd_config_full_parameter_valid" version="1">
+    <ind:subexpression datatype="int" operation="less than or equal">100</ind:subexpression>
+  </ind:textfilecontent54_state>
+
+  <ind:textfilecontent54_test check="all" check_existence="only_one_exists" id="tst_maxstartups_start_parameter" version="1" comment="SSH MaxStartups start parameter is less than or equal to 10">
+    <ind:object object_ref="obj_sshd_config_maxstartups_first_parameter" />
+    <ind:state state_ref="ste_sshd_config_start_parameter_valid" />
+  </ind:textfilecontent54_test>
+  <ind:textfilecontent54_test check="all" check_existence="only_one_exists" id="tst_maxstartups_rate_parameter" version="1" comment="SSH MaxStartups rate parameter is greater than or equal to 30">
+    <ind:object object_ref="obj_sshd_config_maxstartups_second_parameter" />
+    <ind:state state_ref="ste_sshd_config_rate_parameter_valid" />
+  </ind:textfilecontent54_test>
+  <ind:textfilecontent54_test check="all" check_existence="only_one_exists" id="tst_maxstartups_full_parameter" version="1" comment="SSH MaxStartups full parameter is less than or equal to 100">
+    <ind:object object_ref="obj_sshd_config_maxstartups_third_parameter" />
+    <ind:state state_ref="ste_sshd_config_full_parameter_valid" />
+  </ind:textfilecontent54_test>
+
+  <local_variable id="variable_sshd_config_maxstartups" datatype="int"
```

Comment:
@alexhaydock is this done? I see this variable sitting here but I don't actually see it used. :-)

Reply:
I think I was thinking about splitting this into three portions (start/rate/full) and then using it in the `ste_sshd_config_{start,rate,full}_parameter_valid` state checks using the `var_ref` parameter. 

Here, for the `local_variable` I'd use a `variable_component` rather than an `object_component` and simplifying the regex based off of the actual value of the `var_sshd_set_maxstartups` variable.

Reply:
This is missing an `external_variable` definition to actually "pull in" the value of the XCCDF variable.
E.g.: https://github.com/ComplianceAsCode/content/blob/607b04549c5c542042af19fda6b5df6c7a87520d/linux_os/guide/system/selinux/selinux_state/oval/shared.xml#L25

> I think I was thinking about splitting this into three portions (start/rate/full) and then using it in the `ste_sshd_config_{start,rate,full}_parameter_valid` state checks using the `var_ref` parameter.

Once the external variable is available, I'd split the variable into three `local_variables`, with regexes similar to what you do for `obj_sshd_config_maxstartups_{first,second,third}_parameter`.
And then update `ste_sshd_config_{start,rate,full}_parameter_valid` to use a `var_ref` to the `local_variable`s defined above.

> If you don't want to, I think this is OK to merge as-is and I can add tests for this case and update the test to it. Was there a design decision towards this route that I'm missing? :)

Well, at the moment the OVAL check is not using the variable, which is mentioned in the rule description and used in the remediations.
So for consistency, I'd rather update the OVAL to use the variable or remove it from rule and remediation.

Reply:
Yep, exactly what I was thinking about doing. :) If @alexhaydock doesn't want to get to it, I can push one. :-)